### PR TITLE
Combat ui

### PR DESCRIPTION
**Pull request by tigerdong**

### CODE REVIEW

File: characterEntities/package-info.java [hunk 1]
```diff
@@ -1,8 +0,0 @@
-/**
- * 
- */
-/**
- * @author Cathy
- *
- */
-package characterEntities;
```
Comment (CathyWrHua) [inline characterEntities/package-info.java:8]: Not sure why package info was deleted, unless you deleted the entire characterEntity package. These are used for code documentation

File: characterEntities/YellowHero.java [hunk 2]
```diff
@@ -86,4 +56,28 @@ protected boolean isHit(Ability ability, Entity target) {
 		}
 		return hit;
 	}
+
+	@Override
+	public void update() {
+		super.update();
```
Comment (CathyWrHua) [inline characterEntities/YellowHero.java:59]: This update function looks pretty repetitive across all the hero types. Is there a reason why it was refactored this way?

File: screens/GameScreen.java [hunk 3]
```diff
@@ -105,9 +105,9 @@ public void keyPressed(KeyEvent e) {
 		} else if (e.getKeyCode() == KeyEvent.VK_K) {
 			map.setMap(2);
 		} else if (e.getKeyCode() == KeyEvent.VK_A) {
-			player.attack(Hero.Ability.DEFAULT, targets);
+			player.attack(Hero.Ability.DEFAULT);
 		} else if (e.getKeyCode() == KeyEvent.VK_Z) {
-			player.inflict(25, true);
+			damageMarkers.add(player.inflict(25, new Dummy(-100, -100, true)));
```
Comment (CathyWrHua) [inline screens/GameScreen.java]: maybe mark it as a hack

File: characterEntities/package-info.java (hunk 1 above)
Comment (tigerdong) [reply characterEntities/package-info.java:8]: it was empty and useless, just add it back when there's something worth putting in there

File: screens/GameScreen.java (hunk 3 above)
Comment (tigerdong) [reply screens/GameScreen.java]: done

File: screens/GameScreen.java [hunk 4]
```diff
@@ -172,7 +183,10 @@ public void paintComponent(Graphics g) {
 			target.draw(g);
 		}
 		player.draw(g);
-		for(DamageMarker marker : damageMarkers) {
+		for (Animation animation : animations) {
```
Comment (CathyWrHua) [inline screens/GameScreen.java:187]: I still think we can keep the number of arraylists to a minimum, but let's add a todo for refactoring here for now. Once animations are done, we can see how to refactor.

File: characterEntities/YellowHero.java (hunk 2 above)
Comment (tigerdong) [reply characterEntities/YellowHero.java:59]: All the heroes have the same default ability so it's currently repetitive. This function will be different across the heroes in the future

File: screens/GameScreen.java (hunk 4 above)
Comment (tigerdong) [reply screens/GameScreen.java:187]: if we try reducing the size of the arraylists anymore, problems will arise with rendering that will be more of a hassle to deal with than simply having a few arraylists

**Pull request merged**